docs/messages_guide/src/asciidoc/_chapters/ddl_msgs.adoc

```diff
@@ -486,62 +534,52 @@ catalog name and resubmit.
 == SQL 1036
 
 ```
-Only super ID can execute DROP SQL.
+Authorization ID <auth-id-1> cannot grant to authorization ID <auth-id-2> because it could create a circular dependency.
 ```
 
-*Cause:* An attempt was made to drop SQL by a user who is not the super
-ID.
+Where <auth-id-1> is the authorization ID performing the GRANT.
+
+Where <auth-id-2> is the authorization ID that is the target of the GRANT.
+
+*Cause:* There is a chain of grants leading from <auth-id-2> back to <auth-id-1>. Granting
+this request might create a circular dependency.
 
 *Effect:* The operation fails.
 
-*Recovery:* The super ID needs to issue the DROP SCHEMA statement.
+*Recovery:* None.
 
 <<<
 [[SQL-1037]]
 == SQL 1037
 
 ```
-{project-name} is not installed on system <node>.
+Revoke failed because of a dependent grant between authorization ID <auth-id-1> and authorization ID <auth-id-2>.
 ```
 
-Where <node> is the name of the node referenced by the operation.
-
-*Cause:* In a replication or distribution context: An operation
-attempted to create a catalog reference or a partition on a node where
-the {project-name} database has not been installed.
-
-*Effect:* The operation fails.
-
-*Recovery:* Either reissue the statement specifying a different node, or
-install the {project-name} database on <node> and resubmit.
-
-[[SQL-1038]]
-== SQL 1038
-
-```
-Not all user catalogs have been dropped from the system.
-```
+Where <auth-id-1> and <auth-id-2> are authorization IDs.
 
-*Cause:* You attempted to drop SQL while one or more user-created
-catalogs existed.
+*Cause:* You attempted to revoke a privilege from <auth-id-1>, but 
+that user or role has granted that privilege to a third authorization ID.
 
 *Effect:* The operation fails.
 
-*Recovery:* You must drop all user-created catalogs before dropping SQL.
+*Recovery:* None.
 
 <<<
 [[SQL-1039]]
 == SQL 1039
 
 ```
-The DROP SQL statement could not be executed.
+No privileges were revoked. You lack the grant option for the specified privilege(s).
 ```
 
-*Cause:* See the accompanying error message for the cause.
+*Cause:* You attempted to revoke privileges from an authorization ID
+that does not have the grant option for those privileges.
```

Comment:
Cause: you attempted to revoke privileges from an authorization ID but you do not have ability (grant option) to perform this operation.